service/src/main/java/bio/terra/workspace/service/workspace/flight/WorkspaceDeleteFlight.java

```diff
@@ -38,33 +40,46 @@ public WorkspaceDeleteFlight(FlightMap inputParameters, Object applicationContex
     RetryRule cloudRetryRule = RetryRules.cloudLongRunning();
     RetryRule terraRetryRule = RetryRules.shortExponential();
 
-    // In Azure, we need to explicitly delete the controlled resources as there is no containing
-    // object (like a GCP project) that we can delete which will also delete all resources
+    // Delete resources before sam resources - In Azure & AWS, we need to explicitly delete the
+    // controlled resources as there is no containing object (like a GCP project) that we can delete
+    // which will also delete all resources
     addStep(
         new DeleteControlledAzureResourcesStep(
             appContext.getResourceDao(),
             appContext.getControlledResourceService(),
             appContext.getSamService(),
             workspaceUuid,
             userRequest));
+    addStep(
+        new DeleteControlledAwsResourcesStep(
```

Comment:
Is Cloud Context deletion async?  These deletions could take a long time (thus previous comment that Storage Folder delete must be async).  I suspect this has already been encountered in Azure...

Reply:
Looks like workspace and cloud context deletion are sync
WS deletion flow - 

- https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java#L429
- https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java#L370

delete job submit-and-wait till it completes. 

and within the deleteWorkspaceFlight 
- https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/workspace/flight/WorkspaceDeleteFlight.java#L44
Azure deletion functions internally call the deleteSync functions 

1) Just deleting the cloud context deletes the DB entries (not the resources). Is there any use case for this? 
2) ideally the entire workspace deletion must be async. I can create a new task for this, it touches GCP & Azure and will be a larger changeset.

Reply:
We should definitely not take this on in this PR.  Let's keep workspace deletion as you've written it, and medium term think about making this async (a ticket might already exist for this in Broad Jira).  If AWS workspace deletion gives us timeout trouble in the meantime we can maybe do some soft enforcement in Axon UI/CLI.

Reply:
created https://verily.atlassian.net/browse/TERRA-502